code/components/extra-natives-five/src/FileNatives.cpp

```diff
@@ -0,0 +1,92 @@
+#include <StdInc.h>
+#include <Hooking.h>
+
+#include <ScriptEngine.h>
+#include <fiDevice.h>
+
+#include <botan/sha2_32.h>
+#include <boost/algorithm/string.hpp>
+
+static const std::array<std::string_view, 7> kWhitelistedPaths = {
+	"platform:/",
+	"common:/",
+	"update:/",
+	"update2:/",
+	"audio:/",
+	"commoncrc:/",
+	"platformcrc:/"
+};
+
+static bool IsPathWhitelisted(std::string_view path)
+{
+	return std::any_of(kWhitelistedPaths.begin(), kWhitelistedPaths.end(),
+	[&path](const auto& prefix)
+	{
+		return boost::algorithm::starts_with(path, prefix);
+	});
+}
+
+static std::string ToHexString(const uint8_t* data, size_t length)
+{
+	static const char hex_chars[] = "0123456789ABCDEF";
+	std::string result(length * 2, ' ');
+
+	for (size_t i = 0; i < length; ++i)
+	{
+		result[2 * i] = hex_chars[data[i] >> 4];
+		result[2 * i + 1] = hex_chars[data[i] & 0xF];
+	}
+
+	return result;
+};
+
+static HookFunction hookFunction([]()
+{
+	fx::ScriptEngine::RegisterNativeHandler("GET_MOUNTED_FILE_HASH", [](fx::ScriptContext& context)
+	{
+		const char* path = context.CheckArgument<const char*>(0);
+		std::string result = "missing";
+
+		if (IsPathWhitelisted(path))
+		{
+			rage::fiDevice* device = rage::fiDevice::GetDevice(path, true);
+
+			if (device)
+			{
+				auto handle = device->Open(path, true);
+
+				if (handle != -1)
+				{
+					auto len = device->GetFileLength(handle);
+					std::vector<char> data(len);
+
+					device->Read(handle, data.data(), len);
+					device->Close(handle);
+
+					std::vector<uint8_t> fixedHeader(data.begin(), data.begin() + 16 * (data.size() / 16));
+					for (size_t i = 0; i < fixedHeader.size(); i += 16)
+					{
+						auto x = *(uint32_t*)(&fixedHeader[i + 4]);
+						if (x != 0x7fffff00 && (x & 0x80000000) == 0)
+						{
+							auto y = (uint32_t*)(&fixedHeader[i + 12]);
+							if (*y)
+							{
+								*y = 1;
+							}
+						}
+					}
+
+					Botan::SHA_256 hashFunction;
+					hashFunction.update(fixedHeader);
+					std::vector<uint8_t> hash(hashFunction.output_length());
+					hashFunction.final(hash.data());
+
+					result = ToHexString(hash.data(), hash.size());
+				}
+			}
+		}
+
+		context.SetResult(result.c_str());
```

Comment:
SetResult doesn't copy the string, which will be destroyed once the native returns. You need to make `result` static so its contents lasts long to be copied.